test/views/helpers/CalcBreakdownHelperSpec.scala

```diff
@@ -0,0 +1,582 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.helpers
+
+import assets.BaseTestConstants._
+import assets.CalcBreakdownTestConstants.{busPropBRTCalcDataModel, _}
+import assets.EstimatesTestConstants._
+import assets.IncomeSourceDetailsTestConstants._
+import assets.Messages
+import auth.MtdItUser
+import config.FrontendAppConfig
+import implicits.ImplicitCurrencyFormatter._
+import models.calculation.CalculationDataModel
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.i18n.Messages.Implicits._
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import testUtils.TestSupport
+
+
+class CalcBreakdownHelperSpec extends TestSupport {
+
+  lazy val mockAppConfig: FrontendAppConfig = app.injector.instanceOf[FrontendAppConfig]
+
+  val bizAndPropertyUser: MtdItUser[_] = MtdItUser(testMtditid, testNino, Some(testUserDetails), businessAndPropertyAligned)(FakeRequest())
+  val bizUser: MtdItUser[_] = MtdItUser(testMtditid, testNino, Some(testUserDetails), singleBusinessIncome)(FakeRequest())
+  val propertyUser: MtdItUser[_] = MtdItUser(testMtditid, testNino, Some(testUserDetails), propertyIncomeOnly)(FakeRequest())
+
+
+  private def pageSetup(calcDataModel: CalculationDataModel, user: MtdItUser[_]) = new {
+    lazy val page: HtmlFormat.Appendable =
+      views.html.helpers.calcBreakdownHelper(calculationDisplaySuccessModel(calcDataModel), None, testYear)(FakeRequest(), applicationMessages, mockAppConfig, user)
+    lazy val document: Document = Jsoup.parse(contentAsString(page))
+
+    implicit val model: CalculationDataModel = calcDataModel
+
+    def personalAllowanceTotal: String = "-" + (
+      model.personalAllowance +
+        model.savingsAndGains.startBand.taxableIncome +
+        model.savingsAndGains.zeroBand.taxableIncome
+      ).toCurrencyString
+  }
+
+
+  "The Calculation Breakdown View" should {
```

Comment:
rename to "The  Calculation Breakdown VIew on the Estimate Page"

Reply:
Incorporated the review comments
Thanks